src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1322,7 +1742,91 @@ int fmi3_xml_handle_StringVariable(fmi3_xml_parser_context_t *context, const cha
     return 0;
 }
 
-fmi3_xml_enum_variable_props_t * fmi3_xml_parse_enum_properties(fmi3_xml_parser_context_t* context, fmi3_xml_enum_variable_props_t* declaredType) {
+/**
+ * Placeholder handler to initialize the framework.
+ * 
+ * One of the Variable type-specific handlers should always be used.
+*/
+int fmi3_xml_handle_Start(fmi3_xml_parser_context_t* context, const char* data) {
+    assert(0);
+    return 0;
+}
+
+/**
+ * Scans a hexstring to a bytearray.
+ *
+ * @param hexstr The input string to scan.
+ * @param bytearr (Output arg.) A buffer which must be able to fit the scanned bytes.
+*/
+static int fmi3_xml_hexstring_to_bytearray(fmi3_xml_parser_context_t* context, const char* hexstr, uint8_t* bytearr) {
+    size_t len = strlen(hexstr);
+    if (len % 2 != 0) {
+        // Note: 2 hexadecimal characters represent 1 byte.
+        fmi3_xml_parse_error(context, "Hexadecimal string is not of even length: %s", hexstr);
+        // TODO: Or just warning and pad with trailing '0'?
+        return -1;
+    }
+
+    const char* pos = hexstr;
+    size_t nByte = len / 2;
+    for (size_t i = 0; i < nByte; i++) {
+        for (size_t j = 0; j <= 1; j++) {
+            char ch = pos[j];
+            if (!((ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F'))) {
```

Comment:
Would be sensible/worthwhile to only do this type of evaluation if sscanf fails?